### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by andymond**
* Add permissions for platform_admin, chef, cook & visitors
* Add new edge case tests to make sure users can only access parts of site their roles permit

### CODE REVIEW

File: app/services/permission.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  def visitor_permissions
+    return true if controller == "welcome"
+    return true if controller == "dashboard"
+    return true if controller == "sessions"
+    return true if controller == "gsessions"
+    return true if controller == "restaurants" && action.in?(%w(new create))
+    return true if controller == "google_users"
+    return true if controller == "users" && action.in?(%w(new create))
```
Comment (andymond) [inline app/services/permission.rb]: Not happy about this permission. Need to find another way to invite new users.

**Comment by andymond** [comment]
Visitor permissions need to be controlled further for user account creation
